### PR TITLE
Use the error modal to display any error message on the registration form

### PR DESCRIPTION
**Pull request by yuki24**
This pull request replaces the `status` message added in [the previous PR](https://github.com/artsy/force/pull/5699/files#diff-11bf8a936a98ab5a5856792a42114073R116-R120) with the error modal that was already in place.

## Screenshot

error-modal

### CODE REVIEW

**Comment by yuki24** [comment]
This PR is ready for review.

File: src/v2/Apps/Auction/Routes/Register/index.tsx [hunk 1]
```diff
@@ -97,33 +108,35 @@ export const RegisterRoute: React.FC<RegisterProps> = props => {
...
-            )
-          })
-          .catch(error => {
-            handleMutationError(actions, error)
-          })
-      })
-      .catch(error => {
-        handleMutationError(actions, error)
-      })
+  async function createTokenAndSubmit(
```
Comment (dleve123) [inline src/v2/Apps/Auction/Routes/Register/index.tsx:114]: for my own education – what's the thinking between having these function be declared within the "namespace" of the component that is using them (before this commit) vs just injected the functions into the component, but defining them outside of the component "namespace" as you are now currently doing?

Any particulars of JS/TS to consider? Or is there a general code organization point?

File: src/v2/Apps/Auction/Routes/Register/index.tsx [hunk 2]
```diff
@@ -97,33 +108,35 @@ export const RegisterRoute: React.FC<RegisterProps> = props => {
...
+    values: FormValuesForRegistration,
+    actions: FormikActions<FormValuesForRegistration>
+  ) {
+    const address = toStripeAddress(values.address)
+    const { phoneNumber } = values.address
+    const { setFieldError, setSubmitting } = actions
+
+    try {
```
Comment (dleve123) [inline src/v2/Apps/Auction/Routes/Register/index.tsx:122]: Is it correct to say that this block is the crux of the behavior change introduced by this PR? By only catching errors in one place, we ensure we're handling them a consistent way?

File: src/v2/Apps/Auction/Routes/Register/index.tsx (hunk 1 above)
Comment (yuki24) [reply src/v2/Apps/Auction/Routes/Register/index.tsx:114]: * It creates more refactoring opportunities. For instance, the `CreateCreditCardAndUpdatePhone` mutation was [part of the function component](https://github.com/artsy/reaction/blob/549f190198f16321694cc096d680b6de674957ff/src/Apps/Auction/Routes/Register/index.tsx#L91) and then [extracted](https://github.com/artsy/reaction/commit/e2d6154ba8bdba8c04ec56b06d7a7d0ca2652b7c#diff-db7fa0c494cb2467a50b4b0bbc55391fR40) out of it, and now it sits in [its own file](CreateCreditCardAndUpdatePhone) shared by two different components. Now there is another refactoring opportunity for the `saleConfirmRegistrationPath` function and [the `confirmRegistrationPath ` function](https://github.com/artsy/force/blob/a190b8ddf4d4adc473f6b868b11e12658e78fb0b/src/v2/Apps/Auction/getRedirect.ts#L79-L80).
 * This is micro optimization, but functions within the component get re-defined every time it gets re-rendered. That makes [the JIT compiler](https://v8.dev/blog/jitless) harder to optimize the runtime performance.

File: src/v2/Apps/Auction/Routes/Register/index.tsx (hunk 2 above)
Comment (yuki24) [reply src/v2/Apps/Auction/Routes/Register/index.tsx:122]: Correct. We could make the behaviour consistent without consolidating these two functions, but I think it's simpler when we do both in one function.

**Comment by dleve123** [comment]
Re-assigned to you @yuki24 so that you can do any coordinated deployed needed. Merge at your discretion

**Pull request merged**